src/pykoala/data_container.py

```diff
@@ -248,7 +250,132 @@ class Parameter(object):
     """Class that represents some parameter and associated metadata"""
     def __init__(self) -> None:
         pass
+
+class GenericFlagMap(bitmask.BitFlagNameMap):
+    """PyKOALA generic BitMask flag map.
+    
+    Attributes
+    ----------
+    - CR: Cosmic ray (2)
+    - HP: Hot pixel (4)
+    - DP: Dark pixel (8)
+
+    Example
+    -------
+    Accessing the description of each BitFlag is possible by
+    `GenericFlagMap.FLAG_NAME.__doc__`
+    """
+    CR = 2, 'Cosmic ray'
+    HP = 4, 'Hot pixel'
+    DP = 8, 'Dark pixel'
     
+class DataMask(object):
+    """This class contains mask information of DataContainers.
+    
+    Description
+    -----------
+    This class represents a bitmask of a data structure defined by a given
+    dimensionality shape.
+
+    Attributes
+    ----------
+    - data: (np.ndarray)
+        The mask data values.
+    - flag_map: (astropy.bitmask.BitFlagNameMap)
+        A mapping between the flag names and their numerical values.
+
+    Methods
+    -------
+    - get_flags
+    - add_flags
+    - get_mask
+    """
+    def __init__(self, shape, flag_map=GenericFlagMap):
+        # Initialise the mask with all pixels being valid
+        self.data = np.zeros(shape, dtype=int)
+        self.flag_map = flag_map
+
+    def get_flags(self):
+        """Return the current flags used in the mask."""
+        att = list(self.flag_map.__dict__)
+        return [a for a in att if "_" not in a]
+
+    def get_flag_values(self):
+        """Get the values associated to the flags"""
+        flags = self.get_flags()
+        values = {}
+        for flag in flags:
+            values[flag] = getattr(self.flag_map, flag).real
+        return values
+
+    def get_description(self):
+        """Get the description of the BitFlags."""
+        flags = self.get_flags()
+        doc = {}
+        for flag in flags:
+            doc[flag] = getattr(self.flag_map, flag).__doc__
+        return doc
+
+    def add_flags(self, **extra_flags):
+        """Add a new flag to the mask."""
+        self.flag_map = bitmask.extend_bit_flag_map(
+            "DQ_MASK", self.flag_map, **extra_flags)
```

Comment:
Here your peg is gone ;^D
I guess we don't need this function, but I guess/hope it would complain"DQ_MASK" does not exist...

Reply:
This class has been refactored significantly, aiming for a more simple approach without additional dependencies.